l10n_es_partner/models/l10n_es_partner.py

```diff
@@ -149,18 +132,32 @@ class ResPartner(models.Model):
 
     comercial = fields.Char('Trade name', size=128, select=True)
 
-    def name_search(self, cr, uid, name, args=None, operator='ilike',
-                    context=None, limit=100):
-        if not args:
-            args = []
-        partners = super(ResPartner, self).name_search(cr, uid, name, args,
-                                                       operator, context,
-                                                       limit)
-        ids = [x[0] for x in partners]
-        if name and len(ids) == 0:
-            ids = self.search(cr, uid, [('comercial', operator, name)] + args,
-                              limit=limit, context=context)
-        return self.name_get(cr, uid, ids, context=context)
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        """Include commercial name in direct name search. It serves also
+        implicitly for name_search()"""
+        if args and args[0][0] == 'name':
```

Comment:
Hi @pedrobaeza,

This code introduces regression. In your code if you make a simple search like;
`supplier = partner_obj.search([('name', '=', 'Google'), ('supplier', '=', True)], limit=1)`

It will fail with:
`  args = ['|', ('comercial', exp[1], exp[2])] + args
IndexError: string index out of range`